src/NumericalAlgorithms/LinearOperators/PartialDerivatives.cpp

```diff
@@ -121,45 +124,150 @@ auto logical_partial_derivative(
-      const GET_TENSOR(data) < DataVector, GET_DIM(data),                     \
-      GET_FRAME(data) > &u, const Mesh<GET_DIM(data)>& mesh);                 \
-  template TensorMetafunctions::prepend_spatial_index<                        \
-      GET_TENSOR(data) < DataVector, GET_DIM(data), GET_FRAME(data)>,         \
-      GET_DIM(data), UpLo::Lo,                                                \
-      Frame::ElementLogical >                                                 \
-          logical_partial_derivative(const GET_TENSOR(data) < DataVector,     \
-                                     GET_DIM(data), GET_FRAME(data) > &u,     \
-                                     const Mesh<GET_DIM(data)>& mesh);
+#define INSTANTIATION(r, data)                                                 \
+  template void logical_partial_derivative(                                    \
+      gsl::not_null<                                                           \
+          TensorMetafunctions::prepend_spatial_index<                          \
+              GET_TENSOR(data) < DataVector, GET_DIM(data), GET_FRAME(data)>,  \
+          GET_DIM(data), UpLo::Lo, Frame::ElementLogical>* >                   \
+          logical_derivative_of_u,                                             \
+      gsl::not_null<gsl::span<double>*> buffer,                                \
+      const GET_TENSOR(data) < DataVector, GET_DIM(data),                      \
+      GET_FRAME(data) > &u, const Mesh<GET_DIM(data)>& mesh);                  \
+  template void logical_partial_derivative(                                    \
+      gsl::not_null<                                                           \
+          TensorMetafunctions::prepend_spatial_index<                          \
+              GET_TENSOR(data) < DataVector, GET_DIM(data), GET_FRAME(data)>,  \
+          GET_DIM(data), UpLo::Lo, Frame::ElementLogical>* >                   \
+          logical_derivative_of_u,                                             \
+      const GET_TENSOR(data) < DataVector, GET_DIM(data),                      \
+      GET_FRAME(data) > &u, const Mesh<GET_DIM(data)>& mesh);                  \
+  template TensorMetafunctions::prepend_spatial_index<                         \
+      GET_TENSOR(data) < DataVector, GET_DIM(data), GET_FRAME(data)>,          \
+      GET_DIM(data), UpLo::Lo,                                                 \
+      Frame::ElementLogical >                                                  \
+          logical_partial_derivative(const GET_TENSOR(data) < DataVector,      \
+                                     GET_DIM(data), GET_FRAME(data) > &u,      \
+                                     const Mesh<GET_DIM(data)>& mesh);         \
+  template void partial_derivative<GET_TENSOR(data) < DataVector,              \
+                                   GET_DIM(data), GET_FRAME(data)>::symmetry,  \
```

Comment:
(optional) In these instantiations you can omit template parameters of the function that can be inferred automatically be the compiler (this gets rid of line 228 to 230)

Reply:
It turns out that these are necessary here because the `SymmList` and `IndexList` template parameters aren't able to be inferred from the resultant type of `TensorMetafunctions::prepend_spatial_index<...>`

Reply:
At least, this is my understanding of the compile error that is generated when the explicit template parameters are removed

Reply:
That could well be. Type aliases/metafunctions are a bit weird sometimes with deductions. I don't think it's a bad thing to be a bit more explicit either way :)